apps/system/js/app_chrome.js

```diff
@@ -340,6 +373,67 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    this.pinningPref && this.pinningPref.get(function(enabled) {
+      if (!enabled || !this.app.isBrowser()) {
```

Comment:
I think we also need to check for the page loading. If the page is still loading you shouldn't be able to show the card.